packages/keypair/src/generate-vanity-key-pair.ts

```diff
@@ -0,0 +1,68 @@
+import { createKeyPairSignerFromPrivateKeyBytes, type KeyPairSigner } from '@solana/kit'
+
+export const VANITY_MAX_ATTEMPTS = 20_000_000
+
+export interface GenerateVanityKeyPairOptions {
+  attemptsHardCap?: number
+  caseSensitive?: boolean
+  maxAttempts?: number
+  onAttempt?: (attempts: number) => void
+  prefix?: string
+  suffix?: string
+}
+
+export interface VanityKeyPairResult {
+  attempts: number
+  signer: KeyPairSigner
+}
+
+export async function generateVanityKeyPair({
+  attemptsHardCap = VANITY_MAX_ATTEMPTS,
+  caseSensitive = true,
+  maxAttempts,
+  onAttempt,
+  prefix = '',
+  suffix = '',
+}: GenerateVanityKeyPairOptions): Promise<VanityKeyPairResult> {
+  const sanitizedPrefix = prefix.slice(0, 4)
+  const sanitizedSuffix = suffix.slice(0, 4)
+  const hasPrefix = sanitizedPrefix.length > 0
+  const hasSuffix = sanitizedSuffix.length > 0
+
+  if (!hasPrefix && !hasSuffix) {
+    throw new Error('generateVanityKeyPair requires a prefix or suffix')
+  }
+
+  const normalizedPrefix = caseSensitive ? sanitizedPrefix : sanitizedPrefix.toLowerCase()
+  const normalizedSuffix = caseSensitive ? sanitizedSuffix : sanitizedSuffix.toLowerCase()
+
+  const normalizedHardCap =
+    typeof attemptsHardCap === 'number' && Number.isFinite(attemptsHardCap) && attemptsHardCap > 0
+      ? Math.min(Math.floor(attemptsHardCap), VANITY_MAX_ATTEMPTS)
+      : VANITY_MAX_ATTEMPTS
+
+  const attemptsLimit =
+    typeof maxAttempts === 'number' && Number.isFinite(maxAttempts) && maxAttempts > 0
+      ? Math.min(Math.floor(maxAttempts), normalizedHardCap)
+      : normalizedHardCap
```

Comment:
Can we simplify this? It's a ternary statement that goes over 4 lines. Code is untested but would this work?

```
const normalizedHardCap = Math.min(attemptsHardCap ?? VANITY_MAX_ATTEMPTS, VANITY_MAX_ATTEMPTS)
const attemptsLimit = Math.min(maxAttempts ?? normalizedHardCap, normalizedHardCap)
```